### PR TITLE
waf: cmake min version

### PR DESCRIPTION
**Pull request by guludo**
Hi all,

I'm adding support for requiring a minimum version for cmake and using that for PX4. That way we can avoid errors like the one reported in #4254 .

Regards,
Gustavo Sousa

### CODE REVIEW

File: Tools/ardupilotwaf/cmake.py [hunk 1]
```diff
@@ -358,8 +362,29 @@ def create_cmake_build_task(self, cmake_config, cmake_target):
...
+    if not m:
+        cfg.end_msg(
+            'unable to parse version, build is not guaranteed to succeed',
+            color='YELLOW',
+        )
+    else:
+        version = Utils.num2ver(m.group(0))
+        minver = Utils.num2ver(cfg.env.get_flat('CMAKE_MIN_VERSION'))
+        if version < minver:
+            cfg.fatal('cmake must be at least at version 3.2')
```
Comment (OXINARF) [inline Tools/ardupilotwaf/cmake.py]: You forgot the hardcoded 3.2 right?

**Comment by OXINARF** [comment]
I made one small comment, otherwise LGTM

File: Tools/ardupilotwaf/cmake.py (hunk 1 above)
Comment (guludo) [inline Tools/ardupilotwaf/cmake.py]: Ugh :/
Yep, forgot it. Thanks :-)

**Comment by guludo** [comment]
Removed the leftover hardcoded version on the error message.

**Comment by guludo** [comment]
Applied. Thanks for the review @OXINARF .

**Pull request closed without merging**